common/create-store.js

```diff
@@ -0,0 +1,71 @@
+const vendor = require("common/vendor");
+const {createStore, applyMiddleware, combineReducers} = vendor("redux");
+const thunk = vendor("redux-thunk").default;
+const reducers = require("common/reducers/reducers");
+const {Channel} = require("common/ReduxChannel");
+const parseUrlMiddleware = require("common/parse-url-middleware");
+const loggerMiddleware = require("common/redux-logger");
+const {LOCAL_STORAGE_KEY} = require("common/constants");
+
+/**
+ * rehydrateFromLocalStorage - Fetches initial state from local storage
+ *
+ * @return {obj}  Redux initial state as a plain object
+ */
+function rehydrateFromLocalStorage() {
+  let state = {};
+  try {
+    state = JSON.parse(localStorage.getItem(LOCAL_STORAGE_KEY)) || {};
```

Comment:
is there any way for JSON.parse to be successful but we still choose to have an empty object instead?